boot/kernel_os.go

```diff
@@ -53,7 +53,8 @@ func (bp *coreBootParticipant) SetNextBoot() (rebootRequired bool, err error) {
 }
 
 type coreKernel struct {
-	s snap.PlaceInfo
+	s   snap.PlaceInfo
+	dev Device
```

Comment:
I don't think we should carry a dev at this level, we should probably carry already a bootloader.Options or something like that

Reply:
hmm I guess the reason that's weird to me is that where we create the coreKernel we don't have state about the bootloader and that type of thing in the overlord managers, but what we do have that's readily accessible is the model, which implements boot.Device. What do you think we should use instead?

Reply:
I've pushed an update that adds the whole `bootloader.Options` to `coreKernel`. Could have limited it to just the extract-run-kernel flag, but it makes for fewer switches in the actual `*KernelAssets()` methods.

Reply:
ah I misunderstood your comment, I thought the issue was with passing `dev` to `boot.Kernel`, but you just didn't want `dev` in the struct